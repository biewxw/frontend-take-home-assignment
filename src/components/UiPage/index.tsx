import React, { FC } from 'react';
import { Box } from '@chakra-ui/react';

const UiPage: FC = ({ children }) => (
  <Box
    width="100%"
    height="100%"
    paddingX="4rem"
    paddingTop="5rem"
    paddingBottom="2rem"
    overflow="auto"
  >
    {children}
  </Box>
);

export default UiPage;
