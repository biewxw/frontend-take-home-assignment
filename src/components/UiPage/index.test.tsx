import React from 'react';
import UiPage from './index';

import { render } from '@testing-library/react';

describe('<UiPage />', () => {
  it('should render the page with a sample content', () => {
    const { container } = render(
      <UiPage>That's a sample content for the page</UiPage>
    );

    expect(container).toMatchSnapshot();
  });
});
