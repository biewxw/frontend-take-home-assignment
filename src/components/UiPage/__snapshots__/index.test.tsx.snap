// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UiPage /> should render the page with a sample content 1`] = `
<div>
  <div
    class="css-anx49o"
  >
    That's a sample content for the page
  </div>
</div>
`;
